addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -2732,6 +2732,64 @@ QUnit.module('relational_fields', {
         form.destroy();
     });
 
+    QUnit.test('x2many default_order multiple fields', function (assert) {
+        assert.expect(18);
+
+        this.data.partner.records = [
+            {int_field: 10, id: 1, display_name: "record_form"},
+            {int_field: 10, id: 2, display_name: "record2"},
+            {int_field: 10, id: 3, display_name: "record3"},
+            {int_field: 11, id: 4, display_name: "record4"},
+            {int_field: 11, id: 5, display_name: "record5"},
+            {int_field: 11, id: 6, display_name: "record6"},
+            {int_field: 12, id: 7, display_name: "record7"},
+        ];
+
+        this.data.partner.records[0].p = _.shuffle([2, 3, 4, 5, 6, 7]);
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: '<form>' +
+                        '<field name="p" >' +
+                            '<tree default_order="int_field,id">' +
+                                '<field name="id"/>' +
+                                '<field name="display_name"/>' +
+                                '<field name="int_field"/>' +
+                            '</tree>' +
+                        '</field>' +
+                '</form>',
+            res_id: 1,
+        });
+
+        _.each(form.$('.o_field_x2many_list .o_data_row'), function (row, index_row) {
+            var $row = $(row);
+            var expectedValues = [{
+                    field: 'id',
+                    expected: function () {return '' + (index_row + 2);},
+                }, {
+                    field: 'display_name',
+                    expected: function () {return 'record' + (index_row + 2);},
+                }, {
+                    field: 'int_field',
+                    expected: function () {
+                        if (index_row < 2) return '10';
+                        if (index_row < 5) return '11';
+                        return '12';
+                    },
+                }];
+
+            _.each($row.find('.o_data_cell'), function (cell, index_cell) {
+                var $cell = $(cell);
+                assert.strictEqual($cell.text(), expectedValues[index_cell].expected(),
+                    'The value for the field ' + expectedValues[index_cell].field + ' should be correct, otherwise the list is not ordered as expected');
+            });
```

Comment:
what about something like assert.strictEqual(form.$('.o_data_cell.o_id').text(), '1234567'), with classname o_id on field id in the arch?
And by the way, i would set values for int_field s.t. the expected order isn't 1234567.